databallpy/data_parsers/event_data_parsers/statsbomb_parser.py

```diff
@@ -0,0 +1,507 @@
+    """This function retrieves the event data of a specific match. The x
+    and y coordinates provided have been scaled to the dimensions of the pitch, with
+    (0, 0) being the center. Additionally, the coordinates have been standardized so
+    that the home team is represented as playing from left to right for the entire
+    match, and the away team is represented as playing from right to left.
+
+    Args:
+        events_loc (str): location of the events.json file
+        metadata (Metadata): metadata of the match
+        pitch_dimensions (tuple): dimensions of the pitch.
+
+    Returns:
+        Tuple[pd.DataFrame, dict, Metadata]: the event data of the match, the databallpy_events, and the updated metadata
+    """
+    with open(events_loc, "r", encoding="utf-8") as f:
+        events_json = json.load(f)
+
+    formations = {
+        event["team"]["id"]: str(event["tactics"]["formation"])
+        for event in events_json if event["type"]["id"] == 35
+    }
+    metadata.home_formation = formations[metadata.home_team_id]
+    metadata.away_formation = formations[metadata.away_team_id]
+
+    event_data = {
+        "event_id": [],
+        "databallpy_event": [],
+        "period_id": [],
+        "minutes": [],
+        "seconds": [],
+        "player_id": [],
+        "team_id": [],
+        "outcome": [],
+        "start_x": [],
+        "start_y": [],
+        "datetime": [],
+        "statsbomb_event": [],
+        "statsbomb_event_id": [],
+        "player_name": [],
+        "team_name": [],
+    }
+    
+    databallpy_mapping = {
+        "Pass": "pass",
+        "Shot": "shot",
+        "Dribble": "dribble",
+    }
+
+    shot_events = {}
+    pass_events = {}
+    dribble_events = {}
+    
+    event_mask = [event["type"]["id"] not in [5, 18, 19, 26, 27, 34, 35, 36] for event in events_json]
+    for id, event in enumerate(np.array(events_json)[event_mask]): 
+        event_data["event_id"].append(id)
+        event_data["period_id"].append(event["period"])
+        event_data["minutes"].append(event["minute"])
+        event_data["seconds"].append(event["second"])
+        event_data["player_id"].append(event["player"]["id"])
+        event_data["team_id"].append(event["possession_team"]["id"])
+        if "location" in event.keys():
+            event_data["start_x"].append(event["location"][0]-(pitch_dimensions[0]/2))
+            event_data["start_y"].append(event["location"][1]-(pitch_dimensions[1]/2))
+        else:
+            event_data["start_x"].append(MISSING_INT)
+            event_data["start_y"].append(MISSING_INT)
+        event_data["datetime"].append(
+            pd.to_datetime(metadata.periods_frames["start_datetime_ed"][event["period"]-1]+pd.to_timedelta(event["minute"]*60+event["second"], unit="seconds"))
+        )
+        event_data["statsbomb_event"].append(event["type"]["name"].lower().replace("*",""))
+        event_data["statsbomb_event_id"].append(event["id"])
```

Comment:
Do we need two different ids? I know we also did this with opta, but that was because the opta ids were not unique, so we had to create our own unique ones. Is this also the case for statsbomb?